include/openthread/dataset.h

```diff
@@ -125,14 +125,37 @@ struct otPskc
  */
 typedef struct otPskc otPskc;
 
+#define OT_THREAD_1_1_SECURITY_POLICY_FLAGS_LENGTH 1 ///< The Thread 1.1 Security Policy Flags length.
+#define OT_THREAD_1_2_SECURITY_POLICY_FLAGS_LENGTH 2 ///< The Thread 1.2 Security Policy Flags length.
+
 /**
  * This structure represent Security Policy.
  *
  */
 typedef struct otSecurityPolicy
 {
-    uint16_t mRotationTime; ///< The value for thrKeyRotation in units of hours
-    uint8_t  mFlags;        ///< Flags as defined in Thread 1.1 Section 8.10.1.15
+    /*
+     * The value for thrKeyRotation in units of hours.
+     *
+     */
+    uint16_t mRotationTime;
+
+    /*
+     * The Security Policy Flags as defined in Thread 1.2 Section 8.10.1.15.
+     * For Thread 1.1 or lower, only the first byte is available. For Thread
+     * 1.2 or higher, the available `mFlags` bytes are indicated by `mFlagsLength`.
+     *
+     */
+    uint8_t mFlags[OT_THREAD_1_2_SECURITY_POLICY_FLAGS_LENGTH];
```

Comment:
What do you think of  simplifying the public API/definition for user/caller of OT?
- We can define it here as `uint16_t` `mFlag`
- Update the `OT_SECURITY_POLICY_TOBLE_LINK` and `OT_SECURITY_POLICY_NON_CCM_ROUTERS` to use the higher bits in `uint16_t`
- When dealing with a 1.1 build/device we can leave the extra flags as zero (and ignore them if set by user in a OT API call).
- We then deal with the different flag sizes and 1.1 vs 1.2 inside the core implementation.

Thoughts?

Reply:
We can actually consider defining flags as `bool` bitfields in `otSecurityPolicy` struct and hide/remove the actual bit-flag `OT_SECURITY_POLICY_{}` definitions. Thoughts?

```
typedef struct otSecurityPolicy
{
     uint16_t mRotationTime;  ///< The value for thrKeyRotation in units of hours
     bool mObtainMasterKey : 1;
     bool mNativeCommissioning: 1;
    ...
};
```

I feel this would be even simpler to use by OT API users. Thoughts?

Reply:
Defines as bit-fields sounds good to me! I will try to refactor it.

Reply:
Actually, having those flags as two `uint8_t` can simplifies the NCP implementation. This PR is intended to extend the Security Policy flags to two bytes, is it acceptable to keep current flags representation?

Reply:
We can consider including the flag bit mask in NCP/spinel code and convert the `otSecurityPolicy` flags to one or two byte flag that are appended in spinel message. I know it makes the NCP code a bit messy, but I think it may be worth it to be able hide the complexity from the public OT API and make it easier/simpler for user of OT API.

Dealing with two flag bytes and different `OT_SECURITY_POLICY_{}` values (some of which have the same value (same bit)) may make it a bit confusing for some new OT API user (without knowing the details).

Thoughts?

Reply:
I thought it is not good to expose internal headers/definitions to CLI/NCP but looks like we are already doing so :)

I made the change accordingly but not finish yet. Will re-request your review when it is ready! Thanks @abtink

Reply:
@abtink I made the change, PTAL.